packages/jest/src/jest-configuration-builder.ts

```diff
@@ -1,8 +1,38 @@
 import { Path, resolve } from "@angular-devkit/core";
-import { merge } from 'lodash';
+import { mergeWith } from 'lodash';
 import { CustomConfigResolver } from "./custom-config.resolver";
 import { DefaultConfigResolver } from "./default-config.resolver";
 
+/**
+ * A whitelist of property names that are mean to be concat.
+ */
+const ARRAY_PROPERTIES_TO_CONCAT = [
+  // From Jest Config
+  'setupFilesAfterEnv',
+  // From ts-jest config
+  'astTransformers'
+];
+
+/**
+ * This function checks witch properties should be concat. Early return will
+ * merge the data as lodash#merge would do it. 
+ */
+function concatArrayProperties(objValue: unknown, srcValue: unknown, property: string) {
+  if (!ARRAY_PROPERTIES_TO_CONCAT.includes(property)) {
+    return;
+  }
+
+  if (!Array.isArray(objValue)) {
+    return;
+  }
+  
+  if (!Array.isArray(srcValue)) {
+    // This will make Jest to throw, as it is expecting an array, instead of silently fail.
+    return srcValue;
```

Comment:
I'd throw an explicit error here: "Tried to concat non-array property". Moreover the behavior should be the same for object value as well. If I tried to merge array value into non-array value while I listed it in the ARRAY properties I probably mistaken.

Reply:
But this would have an inconsistent behavior for the properties we don't concat, and replace instead. If we leave Jest to throw, it would throw the same error, regardless is a property we need to concat or not.

Reply:
We don't want the error to be thrown if it's not a property we want to concat. We do want to throw an error if it is a property we want to concat but not an array. So the inconsistency is fine, what am I missing?

Reply:
Nothing. Just that I though it would be better to let Jest to throw, as it is currently happening. But I'll update it.